### PR TITLE
Make bin/setup predictable and repeatable and remove bin/update

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

An issue with our current `bin/setup` is that if it's unintentionally run twice it will reset the DB again, wiping out all and any data one has setup, possibly leaving ES with an incosistent state as well as callbacks are not called, the DB is just truncated.

I set out to create a PR in which the default behavior of `bin/setup` wasn't to destroy the DB content and ended up finding a solution inspired by what the next Rails release is going to do:

1. they will unify `bin/update` and `bin/setup` (removing `bin/update) which are basically the same script, except for the first doing `rails db:migrate` and the second doing `rails db:setup` (the destructive command mentioned above). See https://github.com/rails/rails/commit/6f73a31c0ca4438c164152695152a9c947eae912

1. since we don't have `rails db:prepare` (which runs db:migrate for existing dbs and db:setup for non existing dbs) which is coming down with Rails 6 I replaced it with a shell check. If the `PracticalDeveloper_development` is there, we run the migrations, otherwise we run setup.

1. also cleaned up the file copying from https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/bin/setup.tt

Basically you can run `bin/setup` forever and ever without destroying the DB, all the other commands stay the same.

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -24,16 +24,21 @@ chdir APP_ROOT do
...
   unless File.exist?('config/database.yml')
-    cp 'config/database.yml.sample', 'config/database.yml'
+    FileUtils.cp 'config/database.yml.sample', 'config/database.yml'
   end
 
   puts "\n== Preparing database =="
-  system! 'bin/rails db:setup'
+  db_exists = system('psql -o /dev/null -q -n -c "select 1" PracticalDeveloper_development')
+  if db_exists
+    puts 'bin/rails db:migrate'
```
Comment (citizen428) [inline bin/setup]: `puts`, not `system!`?

Comment (rhymes) [reply bin/setup]: hahha sorry, this slipped in a botched save from the editor! Fixed it a tested it multiple times!

File: bin/setup [hunk 2]
```diff
@@ -24,16 +24,21 @@ chdir APP_ROOT do
 
   puts "\n== Copying sample files =="
   unless File.exist?('config/database.yml')
-    cp 'config/database.yml.sample', 'config/database.yml'
+    FileUtils.cp 'config/database.yml.sample', 'config/database.yml'
   end
 
   puts "\n== Preparing database =="
-  system! 'bin/rails db:setup'
+  db_exists = system('psql -o /dev/null -q -n -c "select 1" PracticalDeveloper_development')
```
Comment (lightalloy) [inline bin/setup]: The database name could potentially be different, set by `ENV['DATABASE_URL']`

File: bin/setup [hunk 3]
```diff
@@ -9,8 +8,9 @@ def system!(*args)
   system(*args) || abort("\n== Command #{args} failed ==")
 end
 
-chdir APP_ROOT do
-  # This script is a starting point to setup your application.
+FileUtils.chdir APP_ROOT do
+  # This script is a way to set up or update your development environment automatically.
+  # This script is idempotent, so that you can run it at anytime and get an expectable outcome.
```
Comment (lightalloy) [inline bin/setup]: Great explanation :+1:

File: bin/setup (hunk 2 above)
Comment (citizen428) [reply bin/setup]: Good point, maybe we could make it an optional argument and default to `PracticalDeveloper_development` if that's missing.

File: bin/setup (hunk 3 above)
Comment (rhymes) [reply bin/setup]: Not mine, copied and pasted it from https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/bin/setup.tt ahhahaa

File: bin/setup (hunk 2 above)
Comment (rhymes) [reply bin/setup]: Sure, let's do it.
